From 38e9c6289f28f069fc9077b45712ed1615a4e3f9 Mon Sep 17 00:00:00 2001
From: Donn P Schneider <donnp.schneider@gmail.com>
Date: Tue, 21 Jul 2026 13:02:30 +0300
Subject: [PATCH 5/5] Add new sections to objcopy

EFI binaries output garbage characters on screen and exit without
performing anything.

This problem started with the 3.0.18 version of gnu-efi. The attached
patch adds the new sections that need to be added to the objcopy in
Make.rules.

See https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1108850

Signed-off-by: Donn P Schneider <donnp.schneider@gmail.com>
Signed-off-by: Virgil Bucoci <virgil.bucoci@windriver.com>
---
 Make.rules | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Make.rules b/Make.rules
index 48f4c65..82b9d98 100644
--- a/Make.rules
+++ b/Make.rules
@@ -70,9 +70,9 @@ ifeq ($(ARCH),loongarch64)
 endif
 
 %.efi: %.so
-	$(OBJCOPY) -j .text -j .sdata -j .data -j .dynamic -j .dynsym \
+	$(OBJCOPY) -j .text -j .sdata -j .data -j .dynamic -j dynsym  -j .rodata \
 		   -j .rel -j .rela -j .rel.* -j .rela.* -j .rel* -j .rela* \
-		   -j .reloc $(FORMAT) $*.so $@
+		   -j .areloc -j .reloc $(FORMAT) $*.so $@
 %.so: %.o
 	$(LD) $(LDFLAGS) $^ -o $@ $(LOADLIBES)
 	# check we have no undefined symbols
-- 
2.47.3

